backend/src/test/java/com/crm/service/impl/CarServiceImplTest.java

```diff
@@ -63,4 +66,96 @@ public void getCarByIdShouldThrowExceptionForNotExistingEntry() {
         verify(carRepository, times(1)).findById(ONE);
         verify(carMapper, times(0)).convertToDto(any());
     }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinLengthLessThenSeventeenCharacters() {
+        //given
+        String VIN_TOO_SHORT = "V455";
+
+        //then
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_TOO_SHORT))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_INVALID_LENGTH);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinLengthMoreThenSeventeenCharacters() {
+        //given
+        String VIN_TOO_LONG = "V455467475455845FDLKPRTFGG";
+
+        //then
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_TOO_LONG))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_INVALID_LENGTH);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinContainCharacter_O() {
+        //given
+        String VIN_WITH_CHAR_O = "VIN4003876543920O";
+
+        //when
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_WITH_CHAR_O))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_ILLEGAL_CHARACTERS);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinContainCharacter_I() {
+        //given
+        String VIN_WITH_CHAR_I = "VIN4003876543920i";
+
+        //when
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_WITH_CHAR_I))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_ILLEGAL_CHARACTERS);
+    }
+
+    @Test
+    public void shouldThrowCarNotFoundForVinContainCharacter_Q() {
+        //given
+        String VIN_WITH_CHAR_Q = "VINq4003876543920";
+
+        //when
+        assertThatThrownBy(() -> carService.getCarByVIN(VIN_WITH_CHAR_Q))
+                .isInstanceOf(CarNotFoundException.class)
+                .hasMessage(ErrorDict.VIN_ILLEGAL_CHARACTERS);
+    }
+
+
+    @Test
+    public void shouldGetCarByVin() {
+        //given
+        String vin = "VWVWVW12345678901";
+
+        //TODO
+        CarEntity carEntity = CarEntity.builder()
+                .id(1L)
+                .registrationNumber("KR12345")
+                .brand("Mustang")
+                .model("GT")
+                .productionYear(2010)
+                .mileage(1000)
+                .description("")
+                .vin(vin)
+                .build();
+
+        CarResponse carResponse = CarResponse.builder()
+                .registrationNumber("KR12345")
+                .brand("Mustang")
+                .model("GT")
+                .productionYear(2010)
+                .mileage(1000)
+                .description("")
+                .vin(vin)
+                .build();
+
+        //when
+        when(carRepository.findCarEntityByVinIgnoreCase(vin)).thenReturn(Optional.of(carEntity));
```

Comment:
This is not even used in the test. Should use Mockito.verify to check if the method was invoked.

Reply:
fixed